### PR TITLE
add minimal travis CI build

### PR DESCRIPTION
**Pull request by mackenziestarr**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,7 @@ $(libtermbox):
 	cd vendor/termbox && ./waf configure && ./waf --targets=termbox_static
 
 update_deps:
-	@git submodule update --init --depth 1
+	@git submodule update --init --remote --recursive
```
Comment (mackenziestarr) [inline Makefile]: when i ran this in the travis debug docker container i got `fatal: dumb http transport does not support shallow capabilities` changing to this fixed it though it looks like travis has some of its own behavior around clone depth and submodules
https://docs.travis-ci.com/user/customizing-the-build/#git-clone-depth
https://docs.travis-ci.com/user/customizing-the-build/#git-submodules

File: phpspy.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include <unistd.h>
 
 #ifdef USE_TERMBOX
-#include <termbox.h>
+#include "termbox.h"
```
Comment (mackenziestarr) [inline phpspy.h:28]: I missed this one, only noticed when using the travis debug docker container locally

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+language: php
+php:
+  - '7.3'
+before_script: make
+script: make test
+after_script: make clean
```
Comment (mackenziestarr) [inline .travis.yml:8]: considered doing
```
jobs:
  include:
    stage: build static phpspy binary
       - script: make
     stage: run tests
       - script: make test
```
but it seemed overkill for our all Make setup

**Comment by adsr** [comment]
Sharp! Can you rebase and see if tests pass? Apologies for conflicts in `Makefile`. I think my version will work.

**Comment by mackenziestarr** [comment]
rebased your changes, I wonder if something about this ubuntu setup is causing the failures. afaik we haven't tested ubuntu only debian and centOS

**Pull request merged**